src/levanter/data/shard_cache.py

```diff
@@ -507,6 +619,32 @@ def __iter__(self):
                     logger.exception("Error while reading from shard cache.")
                     raise e
 
+    def await_finished(self, timeout: Optional[float] = None):
+        return ray.get(self.finished_sentinel(), timeout=timeout)
+
+    def attach_metrics_monitor(self, monitor: MetricsMonitor):
+        if self._broker is None:
+            # TODO: decide what to do about attaching if the cache is already finished
+            # maybe get the final metrics?
+            return
+
+        self._metrics_monitors.append(monitor)
+        if self._monitor_thread is None:
+            self._monitor_thread = threading.Thread(target=self._monitor_metrics)
+            self._monitor_thread.start()
+
+    def _monitor_metrics(self):
+        while True:
+            try:
+                metrics = ray.get(self._broker.updated_metrics.remote())
```

Comment:
Got an error message here:

```
Traceback (most recent call last):
  File "/home/ivanzhou/levanter/src/levanter/data/shard_cache.py", line 642, in _monitor_metrics
    metrics = ray.get(self._broker.updated_metrics.remote())
  File "/home/ivanzhou/miniconda3/envs/levanter/lib/python3.10/site-packages/ray/actor.py", line 1189, in __getattr__
    raise AttributeError(
AttributeError: 'ActorHandle' object has no attribute 'updated_metrics'
Exception in thread Thread-8 (_monitor_metrics):
Traceback (most recent call last):
  File "/home/ivanzhou/miniconda3/envs/levanter/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/home/ivanzhou/miniconda3/envs/levanter/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ivanzhou/levanter/src/levanter/data/shard_cache.py", line 649, in _monitor_metrics
    raise e
  File "/home/ivanzhou/levanter/src/levanter/data/shard_cache.py", line 642, in _monitor_metrics
    metrics = ray.get(self._broker.updated_metrics.remote())
  File "/home/ivanzhou/miniconda3/envs/levanter/lib/python3.10/site-packages/ray/actor.py", line 1189, in __getattr__
    raise AttributeError(
AttributeError: 'ActorHandle' object has no attribute 'updated_metrics'
```

Reply:
I couldn't find the definition of this function in our codebase or online. Is this a custom function?

Reply:
botched merge, PTAL